packages/table/src/TableV10/Cell.tsx

```diff
@@ -14,7 +14,9 @@ export const tdInnerDivClassName = createUniqueClassName('td-inner-div');
  * @deprecated
  */
 interface HeaderCellProps
-  extends HTMLElementProps<'th', HTMLTableHeaderCellElement> {
+  extends React.PropsWithChildren<
+    HTMLElementProps<'th', HTMLTableHeaderCellElement>
```

Comment:
Should we do this inside HTMLElementProps? https://jira.mongodb.org/browse/LG-3409